### DISCUSSION TITLE
Update collection using My Tag with PyRekordbox

### DISCUSSION THREAD

**Discussion by lamiphil** [dylanljones/pyrekordbox#122] [Q&A]
Hi ! 

First of all, great work ! I've taken a look at your docs and oh my, the work that was done here is impressive.

I have a little project in mind. I would like to create a local web server that would display all the custom tags I've set in my Rekordbox. 
Using this web server, I hope to be able to filter my collection based on the tags I've chosen.

I know this sounds silly because Rekordbox already allows this. The thing is, I have a small laptop and when I open the panel to display the different filters, it takes up most of my screen and I cannot see all the tags I've created. This makes it frustrating to filter quickly while DJing. 

I wish to be able to access my local web server via my tablet, which would make it easier to filter my tracks.

Would this module allow for this kind of real time communication and real time filtering of tracks in my collection ?

Thank you !

**Comment by dylanljones** [comment]
Hey!

In principle this should be possible. You would have to implement a server on the device with your Rekordbox collection. Using `pyrekordbox`, you can read the database and handle requests to the server. I am just not sure what exactly you are trying to achieve. If you only want to display information from your Rekordbox collection this should be straightforward. 

An issue would be if you want to update your collection *while* using Rekordbox. This library raises an error if you want to commit to the database while Rekordbox is running to prevent confilcts/corrupted data. But reading is possible even while Rekordbox is open.

Hope that helps!

**Reply by lamiphil** [reply]
Hi.

Thank you for your quicky response.

I indeed was looking to update my rekordbox collection while running Rekordbox.

The process would have been the following :
- open rekordbox 
- open my server
- the server takes all the filtering options available in my rekordbox (My Tags, Genres, Keys, etc.)
- connect to the server from my tablet (client side)
- the client would display all the filtering options

Use case :
- I choose any filtering option displayed on my tablet
- the request is sent to the server
- the server transfers the request to Rekordbox
- my track collection is updated based on the filtering options selected 

Unfortunately, I understand that this is not possible due to the error that is thrown if rekordbox is running while executing the module.

Could there be any work around ?

Thank you again !
